state/factory/workingset.go

```diff
@@ -653,8 +654,14 @@ func (ws *workingSet) pickAndRunActions(
 		blobLimit           = params.MaxBlobGasPerBlock / params.BlobTxBlobGasPerBlob
 		deadline            *time.Time
 		fullGas             = blkCtx.GasLimit
+		hasPutPollResult    = slices.ContainsFunc(postSystemActions, func(e *action.SealedEnvelope) bool {
+			_, ok := e.Action().(*action.PutPollResult)
+			return ok
+		})
 	)
-	if ap != nil {
+	// TODO: remove this if all delegates are updated with correct validation sequence
+	// this is a temporary solution to avoid putpollresult validation failure when exist staking action in same block
+	if ap != nil && !hasPutPollResult {
```

Comment:
does this need to be a hard-fork flag? i mean, is it possible it happened in the past?

Reply:
It has already happened in block 5218561 at least, but a hard fork is not necessary; both old and new delegate can validate it.